### DISCUSSION TITLE
Set branch protection to "default" branches

### DISCUSSION THREAD

**Discussion by andreasciamanna** [community/community#22166] [New to GitHub]
<p>I know how to configure branch protection and about the new wildcards support, but is there a way always to define **master** and **develop** (for instance) as protected branches with the same rules?</p>
<p>The wildcards pattern does not work in this case. I would work if it could support regular expression, but I know that’s not possible at the moment.</p>
<p>We have many projects (and more coming), and they should all follow the same rules for these two main branches as soon as they are created on GitHub.</p>
<p>Having to repeat the same steps every time is a bit frustrating.   </p>
<p>Also, some projects usually start with just **master** and the **develop** branch gets created only after the first release: project admins must remember to add this branch to the protected ones and trust me, it almost never happens before someone realises that the branch got commits “the wrong way”.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You may want to look at the <a href="https://github.com/apps/settings" rel="nofollow noopener">Probot Settings app</a>. It is specifically designed to make standardizing your configuration between repositories very easy, including branch protections.</p>
<p>:rotating_light: <strong>WARNING</strong> :rotating_light: Note that this app inherently <em>escalates anyone with <code>push</code> permissions to the <strong>admin</strong> role</em>, since they can push config settings to the <code>master</code> branch, which will be synced. Use with caution.</p>

**Comment by andreasciamanna** [comment]
<p>That looks interesting but also too dangerous.</p>
<p>I think something native would prevent these permission issues and streamline the project’s configuration.</p>
<p>The drawbacks of this suggestion would be too risky and I’m not sure why this should make the issue “Solved”.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/andreasciamanna">@andreasciamanna</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub, and we consider every suggestion we receive. I’ve logged your feedback in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by Vadorequest** [comment]
<p>I have the exact same need. I’d like to globally protect the “master” branch of any repo within my organization so that it refuses direct push but requires PR instead. I can configure that manually for each repository, but that’s quite a waste of time.</p>
<p>I’d also like to have that automated as I fork other OSS projects into my org.</p>

**Comment by archon810** [comment]
<p>Also really surprised there’s no way to specify default branch protection rules for the organization or account to apply to all new repos. This should definitely be a core feature.</p>

**Comment by 74dodgemonaco** [comment]
<p>Yeah this seems to be a very core type feature that is missing and highly necessary in regulated industries or companies.  I currently have a team that has to do this for every repo and it is pretty irritating.</p>
<p>Consider another vote for this as an organizational level setting</p>

**Comment by StellaAthena** [comment]
<aside class="quote no-group" data-username="Vadorequest" data-post="5" data-topic="1256">
<div class="title">
<div class="quote-controls"></div>
 Vadorequest:</div>
<blockquote>
<p>I’d like to globally protect the “master” branch of any repo within my organization so that it refuses direct push but requires PR instead. I can configure that manually for each repository, but that’s quite a waste of time.</p>
<p>I’d also like to have that automated as I fork other OSS projects into my org.</p>
</blockquote>
</aside>
<p>This is exactly my usecase as well. I don’t want to have to add restrictions manually to each branch. Additionally, I’m worried that other team members who create repos might not remember to do this.</p>

**Comment by ChrisAcrobat** [comment]
_Roughly two years later._
I am also searching for a native way of doing this and I think alike the users before me.
- Don't want to do it manually, afraid of missing.
- Waste of time.
- Does in my book really sounds like a core security feature.

I hope there will be a feature for this quite soon. 🙂

EDIT: This discussion, in my opinion, should be moved to [Code Security](https://github.com/orgs/community/discussions/categories/code-security).
